api/collections/serializers.py

```diff
@@ -0,0 +1,126 @@
+from rest_framework import serializers as ser
+from api.base.serializers import JSONAPISerializer, CollectionLinksField, Link, LinksField
+from website.models import Node
+from framework.auth.core import Auth
+
+
+class CollectionSerializer(JSONAPISerializer):
+    filterable_fields = frozenset(['title'])
+
+    id = ser.CharField(read_only=True, source='_id')
+    title = ser.CharField(required=True)
+    date_created = ser.DateTimeField(read_only=True)
+    date_modified = ser.DateTimeField(read_only=True)
+    modified_by = ser.SerializerMethodField(read_only=True, source='get_modified_by')
+    links = CollectionLinksField({
+        'children': {
+            'related': Link('collections:collection-children', kwargs={'pk': '<pk>'}),
+            'count': 'get_node_count',
+        },
+        'pointers': {
+            'related': Link('collections:collection-pointers', kwargs={'pk': '<pk>'}),
+            'count': 'get_pointers_count',
+        },
+    })
+    properties = ser.SerializerMethodField(help_text='A dictionary of read-only booleans: registration, collection,'
+                                                     'and dashboard. Collections are special nodes used by the Project '
+                                                     'Organizer to, as you would imagine, organize projects. '
+                                                     'A dashboard is a collection node that serves as the root of '
+                                                     'Project Organizer collections. Every user will always have '
+                                                     'one Dashboard')
+
+    class Meta:
+        type_ = 'collections'
+
+    def get_node_count(self, obj):
+        if isinstance(obj, dict) and obj['properties']['smart_folder'] is True:
+            return ''
+        else:
+            auth = self.get_user_auth(self.context['request'])
+            nodes = [node for node in obj.nodes if node.can_view(auth)]
+            return len(nodes)
+
+    def get_pointers_count(self, obj):
+        if isinstance(obj, dict) and obj['properties']['smart_folder'] is True:
+            return ''
+        return len(obj.nodes_pointer)
+
+    def get_modified_by(self, obj):
+        if isinstance(obj, dict) and obj['properties']['smart_folder'] is True:
+            return 'Admin'
+        user = obj.logs[-1].user
+        modified_by = user.family_name or user.given_name
+        return modified_by
+
+    def get_user_auth(self, request):
```

Comment:
I think the import problem has been fixed, so this could totally be moved to base/utils now.